sphinx/extensions.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""
+    sphinx.extensions
+    ~~~~~~~~~~~~~~~~~
+
+    Utilities for Sphinx extensions.
+
+    :copyright: Copyright 2007-2016 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import traceback
+
+from six import iteritems
+
+from sphinx.errors import ExtensionError, VersionRequirementError
+from sphinx.locale import _
+from sphinx.util import logging
+
+if False:
+    # For type annotation
+    from typing import Any, Dict  # NOQA
+    from sphinx.application import Sphinx  # NOQA
+
+
+logger = logging.getLogger(__name__)
+
+
+# list of deprecated extensions. Keys are extension name.
+# Values are Sphinx version that merge the extension.
+EXTENSION_BLACKLIST = {
+    "sphinxjp.themecore": "1.2"
+}  # type: Dict[unicode, unicode]
+
+
+class Extension(object):
+    def __init__(self, name, module, **kwargs):
+        self.name = name
+        self.module = module
+        self.version = kwargs.pop('version', 'unknown version')
+        self.parallel_read_safe = kwargs.pop('parallel_read_safe', None)
+        self.parallel_write_safe = kwargs.pop('parallel_read_safe', True)
```

Comment:
The default values of these two variables for parallel build are different. So I have to describe why they are different here.